src/crl/interactivesessions/shells/remoteshell.py

```diff
@@ -0,0 +1,99 @@
+#
+# Nokia - Confidential
+# Do not use, distribute, or copy without consent of Nokia.
+# Copyright (c) 2021 Nokia. All rights reserved.
+#
+# ////////////////////////////////////////////////////////////
+
+import logging
+import paramiko
+
+from .paramikospawn import ParamikoSpawn
+
+from .bashshell import BashShell
+from .registershell import RegisterShell
+from .msgreader import MsgReader
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+@RegisterShell()
+class RemoteShell(BashShell):
+    """
+    This class can be used in order to start a remote bash shell. See also
+    :class:`.BashShell`.
+    **Args:**
+    *ip*: IP address of the host
+    *username*: login username
+    *password*: login passowrd. If not given, passwordless login is expected.
+    *tty_echo*: If True, then terminal echo is set on.
+    *port*: If *port* is not None, alternate port is used in the connection
+            instead of the detfault 22.
+    *init_env*: Path to initialization file which is sourced after the all
+                other initialization is done.
+    *source*: File to source to initial the terminal at the ssh host
+    *sudo* If super user should be used for commands set this to true
+           Default is false. It will initialize shell session with sudo -i
+    For setting timeout for reading login banner, i.e. message-of-day, please use
+    :meth:`.msgreader.MsgReader.set_timeout`.
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, ip, username=None, password=None, tty_echo=False,
+                 port=None, init_env=None, key_file=None, sudo=None):
+        super(RemoteShell, self).__init__(tty_echo=tty_echo, init_env=init_env)
+        self.ip = ip
+        self.username = username
+        self.password = password
+        self.port = port
+        self.key_file = key_file
+        self.sudo = sudo
+        self.ssh = None
+        self.chan = None
+
+    def spawn(self, timeout=20):
+        LOGGER.debug('Spawning RemoteShell using ParamikoSpawn')
+        self.ssh = paramiko.SSHClient()
+        self.ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        self.ssh.connect(self.ip,
+                         username=self.username,
+                         password=self.password,
+                         port=22 if self.port is None else self.port,
+                         key_filename=None if self.key_file is None else self.key_file)
+        self.chan = self.ssh.invoke_shell()
+        LOGGER.debug('Connection channel: %s', self.chan)
+        return ParamikoSpawn(self.chan, timeout=timeout)
+
+    def start(self):
+        self.check_start_success()
+        reader = MsgReader(self._read_until_end)
+        retval = reader.read_until_end()
+        self._sudo_if_needed()
+        retval += self._set_bash_environment()
+        return retval
+
+    def _sudo_if_needed(self):
+        if self.sudo:
+            self._terminal.sendline("sudo -i")
+            self._read_until('sudo -i', timeout=20)
```

Comment:
This does not necessarily mean that sudo shell is started.  This just reads echo of the command.